### PR TITLE
[Inductor] Wrap signed int arithmetic in CPU codegen

### LINKED ISSUE #190764: Equivalent int32 graphs disagree under static CPU Inductor: factorized and PolyJuice max endpoints pass, expanded intermediate fails
### 🐛 Describe the bug

An equality-saturation test generated several mathematically equivalent `int32` graphs. All graphs produce the same eager output, but static CPU Inductor returns an incorrect value only for a strict intermediate graph.

The factorized minimum endpoint passes. A PolyJuice-native most-complex endpoint also passes. The fully expanded intermediate state fails consistently.

This is a silent correctness issue: endpoint-only testing misses the failure, while executing the intermediate equivalent state exposes it.

### Equivalent graph states

All six inputs are `int32[2, 2]` tensors generated from the same edge-value profile.

#### Minimum endpoint — cost 12

```python
def minimum_graph(x, y, z, p, q, r):
    left = x @ (y + z)
    right = p @ (q + r)
    return (left + right).sum(dim=1)
```

#### Strict intermediate state — cost 14

```python
def intermediate_graph(x, y, z, p, q, r):
    left = x @ y + x @ z
    right = p @ q + p @ r
    return (left + right).sum(dim=1)
```

The transformation is matrix-multiplication linearity:

```text
X @ (Y + Z)  ==  X @ Y + X @ Z
```

#### Finite-path terminal state — cost 16

```python
def terminal_graph(x, y, z, p, q, r):
    left = x @ y + x @ z
    right = p @ q + p @ r
    value = left + right
    first, second = torch.chunk(value, 2, dim=0)
    restored = torch.cat((first, second), dim=0)
    return restored.sum(dim=1)
```

The `chunk`/`cat` wrapper is an identity because the same axis and original order are preserved.

### Results

```text
                              eager       static CPU Inductor
minimum endpoint, cost 12     [0, 7]      [0, 7]
intermediate, cost 14         [0, 7]      [0, -4294967289]
terminal state, cost 16       [0, 7]      [0, 7]
```

Each state was independently compiled twice and the results were stable.

### PolyJuice native endpoint extraction

The same seed graph was saturated with PolyJuice's active equivalence rules and evaluated using its native minimum and most-complex extractors.

A semantically invalid transpose/matmul operand ordering in one rule was corrected to the standard identity:

```text
transpose(X @ Y) == transpose(Y) @ transpose(X)
```

All other active rules, saturation limits, and the native extraction algorithms were unchanged.

Across three budgets:

```text
budget              minimum endpoint   intermediate       native maximum endpoint
2 / 100 / 3 s       cost 12, pass      cost 14, mismatch  cost 36, pass
4 / 1000 / 10 s     cost 12, pass      cost 14, mismatch  cost 135, pass
8 / 10000 / 30 s    cost 12, pass      cost 14, mismatch  cost 291, pass
```

For every graph above, eager mode returned `[0, 7]`. Static CPU Inductor returned `[0, -4294967289]` only for the expanded intermediate graph.

### Reduced reproducer from the failing intermediate

The failing intermediate was minimized to the following add-then-reduce pattern:

```python
import torch


def function(left, right):
    return (left + right).sum(dim=1)


left = torch.tensor(
    [[torch.iinfo(torch.int32).max]],
    dtype=torch.int32,
)
right = torch.tensor([[1]], dtype=torch.int32)

expected = function(left, right)
compiled = torch.compile(
    function,
    backend="inductor",
    dynamic=False,
)
actual = compiled(left, right)

print("torch:", torch.__version__)
print("eager:", expected)
print("compiled:", actual)
```

Observed:

```text
eager:    tensor([-2147483648])
compiled: tensor([2147483648])
```

The difference is exactly `2**32`.

### Expected behavior

All eager-equivalent graphs should remain equivalent after compilation.

The `int32` addition should wrap before `sum` promotes the result to `int64`:

```text
2147483647 + 1 -> -2147483648
```

Static CPU Inductor should therefore return the same value as eager mode for both the endpoint and intermediate graph structures.

### Isolation

```text
torch.compile(..., backend="inductor", dynamic=False): fails
torch.compile(..., backend="inductor", dynamic=True):  passes
torch.compile(..., backend="aot_eager"):               passes
```

This points to the static CPU Inductor code-generation or optimization path rather than Dynamo capture or the graph equivalence itself.

### Generated C++ observation

A recent nightly generated code of this form:

```cpp
auto tmp0 = in_ptr0[0];
auto tmp1 = in_ptr1[0];
auto tmp2 = int32_t(tmp0 + tmp1);
auto tmp3 = c10::convert<int64_t>(tmp2);
```

If `tmp0 + tmp1` is evaluated as signed 32-bit arithmetic before the explicit cast, it can invoke C++ signed-overflow undefined behavior. This is a suspected explanation based on the generated code; the final root cause and fix location need maintainer confirmation.

### Reproduced versions

```text
2.5.1
2.12.1
2.13.0
2.14.0.dev20260718
```

### Environment

```text
OS: macOS 26.0
Architecture: Apple Silicon arm64
Python: 3.12.10
Backend: CPU Inductor
Compile mode: static, dynamic=False
Input dtype: int32
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01 @chauhang @penguinwu @voznesenskym @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by guan404ming**
## Related Issue

Fixes #190764

## Why

- Compiled int32 arithmetic returned a value `2**32` away from eager
- Two mathematically equivalent graphs disagreed after compilation

## How

- Do int32/int64 add/sub/mul in the unsigned type, then cast back
- int8/int16 are left alone: they promote to `int`, which cannot overflow at those widths
- Other signed-overflow sites (`neg`, `abs`, `truncdiv`, `reduction_combine`, vec ops) are deliberately out of scope
- `-fwrapv` would cover all of them, but there is no flag hook in `cpp_builder.py` and MSVC has no equivalent

## Test

```
python test/inductor/test_cpu_repro.py -k signed_int_overflow_wraps
python test/inductor/test_cpu_repro.py
```

int32 is the dtype that reproduces; the test widens to int64 so the unwrapped value stays observable, and runs every simdlen.

This PR was authored with the assistance of Claude.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by guan404ming** [comment]
Dropped int8/int16, folded the helper, widened the test to expose wrapping.

**Review by leslie-fang-intel — APPROVED** [review]

**Comment by guan404ming** [comment]
Thanks!
